### ISSUE TITLE
Helix crashes on closing a split window that is currently focused

### ISSUE THREAD

**Issue by js-ojus** [helix-editor/helix#3654]
### Summary

Helix crashes reproducibly upon splitting the window vertically, and attempting to close the newly-created window using `space w q`.

### Reproduction Steps

 

I tried this:

1. `hx a.go`
2. `space w v`
3. `space w q`

I expected this to happen:

The newly-created vertical window on the right should close, leaving behind the original window with the same file `a.go`.

Instead, this happened:

Helix crashed with the following full stack trace.

```
thread 'main' panicked at 'invalid HopSlotMap key used', helix-view/src/tree.rs:288:20
stack backtrace:
   0:        0x109ca26a7 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::hf40d0caef3ef003c
   1:        0x1094a59aa - core::fmt::write::h7fafa4d8eb87db06
   2:        0x109c7e30c - std::io::Write::write_fmt::he2d3473829bfad58
   3:        0x109c84531 - std::panicking::default_hook::{{closure}}::h2e54ef15ea4323a2
   4:        0x109c841ae - std::panicking::default_hook::ha4f40c368619ac79
   5:        0x109c84d4e - std::panicking::rust_panic_with_hook::hd7ebe5d3a09e3b75
   6:        0x109a1b15b - std::panicking::begin_panic::{{closure}}::h5a53d69a3a4fe805
   7:        0x109a1b067 - std::sys_common::backtrace::__rust_end_short_backtrace::hba245a790c2c0cdd
   8:        0x109d5a77d - std::panicking::begin_panic::h9ec0d287bf273668
   9:        0x109842aff - <helix_term::ui::editor::EditorView as helix_term::compositor::Component>::handle_event::hdf43981ab2c2418a
  10:        0x109838ba6 - helix_term::compositor::Compositor::handle_event::hbb9d98a5dc60d7bb
  11:        0x1098570e3 - helix_term::application::Application::handle_terminal_events::h20eaad39b4367ec3
  12:        0x109b03589 - helix_term::application::Application::event_loop_until_idle::{{closure}}::heee3f3c982a47b7a
  13:        0x109b1fbfe - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hc9f93d29fab4c238
  14:        0x109b0f650 - std::thread::local::LocalKey<T>::with::hcd873df379b43be3
  15:        0x109aea4d4 - tokio::park::thread::CachedParkThread::block_on::h1322ebd9f4112065
  16:        0x109b5c9e7 - tokio::runtime::thread_pool::ThreadPool::block_on::h63352f981052ce41
  17:        0x109b2e53f - tokio::runtime::Runtime::block_on::h42f8802dcb14dee1
  18:        0x109aece99 - hx::main::hc57a814645e585ed
  19:        0x109ae42b6 - std::sys_common::backtrace::__rust_begin_short_backtrace::he77a05876416c97f
  20:        0x109b0f0fc - std::rt::lang_start::{{closure}}::h6d5f49781b942269
  21:        0x109c75665 - std::rt::lang_start_internal::he523a7cb5789bc85
  22:        0x109aecfa9 - _main
```

### Helix log

<details><summary></summary>

```
```

</details>

### Platform

macOS

### Terminal Emulator

kitty 0.26.1

### Helix Version

helix 22.08 (ee94031f)

**Comment by A-Walrus** [comment]
This was fixed in a new patch. Try 22.08.1

**Comment by js-ojus** [comment]
Glad to know that, @A-Walrus . I installed Helix through `brew`, which still has 22.08. I shall certainly test it once the patch becomes available.

By the way, I came across Helix only today. I have been testing it on a few of my files. So far, it looks very good! My compliments to all the contributors on creating a very a good editor that looks like it can legitimately compete with `vim`! 👏

**Comment by the-mikedavis** [comment]
Duplicate of https://github.com/helix-editor/helix/issues/3626

**Comment by the-mikedavis** [comment]
It looks like the brew formula update was just merged https://github.com/Homebrew/homebrew-core/pull/109423

**Comment by js-ojus** [comment]
> It looks like the brew formula update was just merged [Homebrew/homebrew-core#109423](https://github.com/Homebrew/homebrew-core/pull/109423)

Yes, I have upgraded and can confirm that this issue is no longer reproducible. Thanks for the quick responses!
